rcl/src/rcl/wait.c

```diff
@@ -64,6 +64,7 @@ static void
 __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
 {
   rcl_ret_t ret;
+  (void)ret;  // NO LINT
```

Comment:
I am argueing for the smaller scope based on out programming conventions (https://github.com/ros2/ros2/wiki/Developer-Guide#programming-conventions):

> Declare variables in the narrowest scope possible.

If you think the checks are fine for debug builds only that is ok for me.